src/ucs/config/parser.c

```diff
@@ -1374,6 +1440,11 @@ ucs_status_t ucs_config_parser_fill_opts(void *opts, ucs_config_field_t *fields,
         goto err_free;
     }
 
+    status = ucs_config_parse_config_file(UCX_CONF_FILE, 0);
+    if (status != UCS_OK) {
+        goto err_free;
+    }
```

Comment:
maybe parse config file on demand, if cannot find a value for some key? 
something like
```
ucs_config_get_value_from_config_file():
  iter = kh_get(..)
 if (iter == end) {
    parse file
    goto again
 }
 ...
}
```

Reply:
" cannot find a value for some key" - you mean in env vars? But that's the "happy flow" - we obviously don't expect each and every possible config to be in env vars, so in practice the file will be always parsed.

By parsing on demand, do we improve the code flow itself, or its readability?

Reply:
i mean in the key/value hash which stores all config file entries